include/ktest.h

```diff
@@ -2,16 +2,59 @@
 #define _SYS_KTEST_H_
 
 #include <linker_set.h>
+#include <stdc.h>
 
-#define TEST_NAME_MAX 32
+#define KTEST_NAME_MAX 32
+
+#define KTEST_SUCCESS 0
+#define KTEST_FAILURE 1
+
+/* Signifies that a test does not return. */
+#define KTEST_FLAG_NORETURN 0x01
+/* Signifies that a test irreversibly breaks internal kernel state, and any
+   further test done without restarting the kernel will be inconclusive. */
+#define KTEST_FLAG_DIRTY 0x02
+/* Indicates that a test enters usermode. */
+#define KTEST_FLAG_USERMODE 0x04
+/* Excludes the test from being run in auto mode. This flag is only useful for
+   temporarily marking some tests while debugging the testing framework. */
+#define KTEST_FLAG_BROKEN 0x08
 
 typedef struct {
-  const char test_name[TEST_NAME_MAX];
+  const char test_name[KTEST_NAME_MAX];
   int (*test_func)();
+  uint32_t flags;
 } test_entry_t;
 
-#define KTEST_ADD(name, func)                                                   \
-  test_entry_t name##_test = {#name, func};                                    \
+void ktest_main(const char *test);
+
+#define KTEST_ADD_FLAGS(name, func, flags)                                     \
```

Comment:
I've got confused while reading tests – I though that you have another linker set for test flags.

Couldn't we just use `KTEST_ADD` with extra argument for flags – it's just extra 3 characters?